recipes/cppast/0001-do-not-vendor-external-dependencies.patch

```diff
@@ -0,0 +1,1463 @@
+-    }
+-    else
+-    {
+-        // first search in current directory, then in PATH
+-        static const char* paths[]
+-            = {"./clang++",   "clang++",       "./clang++-4.0", "clang++-4.0", "./clang++-5.0",
+-               "clang++-5.0", "./clang++-6.0", "clang++-6.0",   "./clang-7",   "clang-7",
+-               "./clang-8",   "clang-8",       "./clang-9",     "clang-9",     "./clang-10",
+-               "clang-10",    "./clang-11",    "clang-11"};
+-        for (auto& p : paths)
+-            if (is_valid_binary(p))
+-            {
+-                clang_binary_ = p;
+-                add_default_include_dirs(*this);
+-                return false;
+-            }
+-
++    if (!is_valid_binary(binary))
+         throw std::invalid_argument("unable to find clang binary '" + binary + "'");
+-    }
++
++
++    clang_binary_ = binary;
++    add_default_include_dirs(*this);
++    return true;
+ }
+ 
+ void libclang_compile_config::do_set_flags(cpp_standard standard, compile_flags flags)
+diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
+index 3b24dac..dc1d6d8 100644
+--- a/test/CMakeLists.txt
++++ b/test/CMakeLists.txt
+@@ -2,12 +2,6 @@
+ # SPDX-License-Identifier: MIT
+ # found in the top-level directory of this distribution.
+ 
+-# Fetch catch.
+-message(STATUS "Fetching catch")
+-include(FetchContent)
+-FetchContent_Declare(catch URL https://github.com/catchorg/Catch2/archive/v2.13.4.zip)
+-FetchContent_MakeAvailable(catch)
+-
+ set(tests
+         code_generator.cpp
+         cpp_alias_template.cpp
+@@ -43,7 +37,7 @@ endforeach()
+ 
+ add_executable(cppast_test test.cpp test_parser.hpp ${tests})
+ target_include_directories(cppast_test PRIVATE ${CMAKE_CURRENT_LIST_DIR}/../src ${CMAKE_CURRENT_BINARY_DIR})
+-target_link_libraries(cppast_test PUBLIC cppast Catch2)
++target_link_libraries(cppast_test PUBLIC cppast)
+ target_compile_definitions(cppast_test PUBLIC CPPAST_INTEGRATION_FILE="${CMAKE_CURRENT_SOURCE_DIR}/integration.cpp"
+                                               CPPAST_COMPILE_COMMANDS="${CMAKE_BINARY_DIR}")
+ 
+diff --git a/tool/CMakeLists.txt b/tool/CMakeLists.txt
+index 5dfa27a..f228e83 100644
+--- a/tool/CMakeLists.txt
++++ b/tool/CMakeLists.txt
+@@ -3,5 +3,5 @@
+ # found in the top-level directory of this distribution.
+ 
+ add_executable(cppast_tool main.cpp)
+-target_link_libraries(cppast_tool PUBLIC cppast cxxopts)
++target_link_libraries(cppast_tool PUBLIC cppast)
+ set_target_properties(cppast_tool PROPERTIES CXX_STANDARD 11 OUTPUT_NAME cppast)
+diff --git a/tool/main.cpp b/tool/main.cpp
+index f0f3a1e..866a135 100644
+--- a/tool/main.cpp
++++ b/tool/main.cpp
+@@ -228,8 +228,6 @@ try
+     {
+         std::cout << "cppast version " << CPPAST_VERSION_STRING << "\n";
+         std::cout << "Copyright (C) Jonathan Müller 2017-2019 <jonathanmueller.dev@gmail.com>\n";
+-        std::cout << '\n';
+-        std::cout << "Using libclang version " << CPPAST_CLANG_VERSION_STRING << '\n';
```

Comment:
This information was constructed when looking for libclang in the old cmake setup. We don't have that information as readily available anymore it seems.

Also, since we are linking dynamically, we won't know the exact version of libclang at compile time. So, better to not print this at all.